lib/array.js

```diff
@@ -181,21 +181,31 @@ module.exports = (api) => {
     //   result - array result
   ) => {
     const len = items.length;
-    let i = 0;
     done = api.metasync.cb(done);
+    if (!len) return done;
 
-    function next() {
-      if (i === len) return done();
-      fn(items[i], (err, accepted) => {
-        if (err) return done(err);
-        if (accepted) return done(items[i]);
-        i++;
-        next();
-      });
-    }
+    let count = 0;
+    const output = new Array(len);
+    let finished = false;
```

Comment:
It is better to rename it to `errored`, because at the moment it represents the state when error occurs during any of the function execution.